### PR TITLE
Compatibility with RN Webview postMessage and onMessage

### PR DESCRIPTION
**Pull request by fchasen**
Updates the `onMessage` prop to be compatible with what the current React Native Webview returns and adds the `window.postMessage` from there as well.

Adds a messagingEnabled prop for disabling.

This will allow using wkwebview on iOS and the default webview for Android without changing the communication methods between platforms.

### CODE REVIEW

File: ios/RCTWKWebView/RCTWKWebView.m [hunk 1]
```diff
@@ -123,7 +123,11 @@ -(void)setHideKeyboardAccessoryView:(BOOL)hideKeyboardAccessoryView
 - (void)userContentController:(WKUserContentController *)userContentController didReceiveScriptMessage:(WKScriptMessage *)message
 {
   if (_onMessage) {
-    _onMessage(@{@"name":message.name, @"body": message.body});
+    NSMutableDictionary<NSString *, id> *event = [self baseEvent];
+    [event addEntriesFromDictionary: @{
+      @"data": message.body,
```
Comment (andreyslyusar) [inline ios/RCTWKWebView/RCTWKWebView.m]: `message.name` allow distinguish between message handlers, and probably we should keep it.

Comment (fchasen) [reply ios/RCTWKWebView/RCTWKWebView.m]: makes sense, I'll add it to the event.

Comment (fchasen) [reply ios/RCTWKWebView/RCTWKWebView.m]: Updated

**Comment by aschenkel** [comment]
Hey! is this PR stuck, in progress or ready to merge?

File: ios/RCTWKWebView/RCTWKWebView.m [hunk 2]
```diff
@@ -354,6 +372,26 @@ - (void)webView:(__unused WKWebView *)webView didFailProvisionalNavigation:(__un
 
 - (void)webView:(WKWebView *)webView didFinishNavigation:(__unused WKNavigation *)navigation
 {
+  if (_messagingEnabled) {
```
Comment (oblador) [inline ios/RCTWKWebView/RCTWKWebView.m]: Have you considered using a user script here instead?

File: ios/RCTWKWebView/RCTWKWebViewManager.m [hunk 3]
```diff
@@ -45,6 +45,8 @@ - (UIView *)view
 RCT_EXPORT_VIEW_PROPERTY(onProgress, RCTDirectEventBlock)
 RCT_EXPORT_VIEW_PROPERTY(onMessage, RCTDirectEventBlock)
 RCT_EXPORT_VIEW_PROPERTY(hideKeyboardAccessoryView, BOOL)
+RCT_EXPORT_VIEW_PROPERTY(messagingEnabled, BOOL)
```
Comment (oblador) [inline ios/RCTWKWebView/RCTWKWebViewManager.m]: Can't this prop be inferred from the availability of `onMessage` instead?

**Comment by boundaryfree** [comment]
Hi, 
When could this branch be merge to the master branch. Now I also need "postMessage" to webview from RN.

**Comment by csbok** [comment]
I also need this branch. please merge it

**Comment by boundaryfree** [comment]
@csbok , I have merge this branch in [my forked repository](https://github.com/boundaryfree/react-native-wkwebview). Also I published it to npm(package name: react-native-wkwebview-reborn-msg)

**Comment by andreyslyusar** [comment]
I will try to catch up with merging all pull requests during Christmas break. @insraq How we collaborate so I can push changes later to npm ?

**Comment by insraq** [comment]
Just mention me, I will push to npm :-)

**Comment by fchasen** [comment]
I'm happy to make any changes as needed.

This PR was based on RN 0.47, so should make sure it is still compatible with the latest RN webview messaging.

**Comment by puckey** [comment]
looking forward to seeing this merged in

File: ios/RCTWKWebView/RCTWKWebView.m (hunk 2 above)
Comment (fchasen) [reply ios/RCTWKWebView/RCTWKWebView.m]: This is just taken from here: https://github.com/facebook/react-native/blob/2546c95ac8eeca5a0056d34188a10f75a4b21a44/React/Views/RCTWebView.m#L301

**Comment by mmorier** [comment]
Interested by this PR, any chance to rebase it and merge it?

**Comment by fchasen** [comment]
I'll resolve these conflicts asap, I'd appreciate any help available with testing it out.

**Comment by insraq** [comment]
@fchasen If you need write access to the repo, just ping me. Also ping me if you want to release a new npm package.

**Comment by fchasen** [comment]
Thanks for merging and updating the readme and example as well.

**Comment by insraq** [comment]
@fchasen I've merged and rewrite the code to use the message bridge in WkWebView, instead of the hack solution by WebView. I've done some testing and it seems to work fine. You can also test in your app.

**Comment by fchasen** [comment]
Great, thanks - that will work fine as I had previously used the same method https://github.com/fchasen/react-native-wkwebview/blob/d1077537eeb83b9440eadd4839c705b6cd1b9634/ios/RCTWKWebView/RCTWKWebView.m#L389 but must have overwritten it when I merged the latest code in from the RN source.

Do you think there is any need to retain the code to check for the original `window.postMessage`?

**Pull request merged**